Flutter_TDD_Architecture_Course_notes.md

```diff
@@ -1 +1,273 @@
 
+## Notes for Flutter TDD Clean Architecture Course Videos
+
+This course aims to create a Number Trivia App, and will show us how we can implement test driven development to all the applications we want to do.
+
+#### 1. Explanation & Project Structure
+> Video:https://www.youtube.com/watch?v=KjE2IDphA_U&list=PLB6lc7nQ1n4iYGE_khpXRdJkJEp9WOech
+
+In the `first tutorial` an explanation is given of the importance of having all functions tested in our application.  
+It also explains the **clean architecture**.
+All the features of the application will be divided into three folders, **domain**, **data** and **presentation**.
+It is also explained how each of these folders works and what they will contain.
+
+<br />
+
+#### 2. Entities & Use Cases 
+> Video:https://www.youtube.com/watch?v=lPkWX8xFthE&list=PLB6lc7nQ1n4iYGE_khpXRdJkJEp9WOech&index=2
+
+In the `second tutorial` we start by adding the packages of the dependencies to the `pubspec.yaml` file, that will be needed to build the application.
+
+````
+dependencies:
+  flutter:
+    sdk: flutter
+
+  get_it: ^2.0.1
+  
+  # Bloc for state management
+  flutter_bloc: ^0.21.0
+  
+  # Value equality
+  equatable: ^0.4.0
+  
+  # Functional programming thingies
+  dartz: ^0.8.6
+  
+  # Remote API
+  connectivity: ^0.4.3+7
+  
+  http: ^0.12.0+2
+  # Local cache
+  shared_preferences: ^0.5.3+4
+
+  cupertino_icons: ^0.1.2
+
+dev_dependencies:
+  flutter_test:
+    sdk: flutter
+  mockito: ^4.1.0
+  ````
+  
+  It is explained how the **entity** will work in this application.
+  The following explains the function of the **use cases** that will collect data from the repositories.
+  The use cases function in this application will collect the entities and show them in the presentation folder where all the elements that we will use physically in our application are located.
+  
+  
+  #### The either Type
+  
+  The **either** type is shown and explained with the function of being able to represent two types at the same time in this case serves for the errors and for the numbers that we will show.
+In which **L** are the **errors** and **R** the **numbers**.
+
+``Future<Either<Failure, NumberTrivia>>``
+ 
+
+Then it is defined the **failures** that will have an own file.
+
+
+
+> Failures.dart
+
+````
```

Comment:
Three backticks and the `dart` keyword so that there is syntax highlighting. 
At present without the `dart` keyword the code is just _grey_:
https://github.com/dwyl/learn-flutter/blob/20c848baa04b69e44068e9140937a3e12e068122/Flutter_TDD_Architecture_Course_notes.md
![image](https://user-images.githubusercontent.com/194400/74646462-d4b4e280-5171-11ea-8d1d-b7c0011edfa2.png)

Reply:
I understood but as I said above when I was doing the summary every time I tried to use less it gave me a mistake and it did not appear as a code but as a sentence.

Reply:
You appear to be editing in the browser. 
Try editing in Atom and use Markdown Preview. 
It's _much_ faster/easier.